### DISCUSSION TITLE
Naming: Popover vs Non-modal dialog

### DISCUSSION THREAD

**Discussion by SiTaggart** [twilio-labs/paste#491] [General] [answered]
I would prefer it if we named our components based on the accessible name of the control, based on the W3C [WAI-ARIA specification](https://www.w3.org/TR/wai-aria-1.1/).

There are a number of benefits to this including shared terminology between engineering and design, education around accessible concepts and improving decision making when choosing a UI control to perform a specific task based on what it can accessibly communicate to a user.

So that leads to me Popover. It's a name that most folks are reasonably comfortable with but it technically doesn't mean anything, but at the same time depending on who you talk to, it can mean many _different_ things.

I would like us to replace the term Popover in our nomenclatures with non-modal dialog going forward. Namely for the following reasons:

- it maps 1 to 1 to international accessibility specifications
- it explicitly describes what the component can and can't do based on the international specification.
- When sharing a conversation between PM, Design and Engineering, googling the term for further clarification will always result in the same, share understanding.

**Comment by aayushpi** [comment]
It's not a very approachable name. Having a simple name really helps.

I propose using `Popover` and explicitly calling out that its how we refer to a non-modal dialog.

**Reply by serifluous** [reply]
+1

**Comment by richbachman** [comment]
As much as I'm on board with naming components based on the aria spec, I feel like this one comes with a long history of being called a popover by both engineers and designers. Even design systems we look up to reference it as a popover, e.g. Chakra, Sancho, and Reakit.

I like the idea of calling out that we refer to it as a non-modal dialog. That way we can educate our users on proper terminology.

**Comment by corinne-nagel** [comment, marked as answer]
Hey all, it looks like popover is the consensus to help keep it approachable/simple but calling it out somewhere that we also refer to it as a non-modal dialog. I'm going to consider this answered unless anybody has any hesitations still. Thanks!
